File: app/lib/models/traveler.dart
class Traveler {
  final List<String> preferredRegions;
  final List<String> travelStyle;
  final DateTime createdAt;
  final DateTime updatedAt;

  Traveler({
    required this.preferredRegions,
    required this.travelStyle,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Traveler.fromJson(Map<String, dynamic> json) => Traveler(
        preferredRegions: List<String>.from(json['preferred_regions']),
        travelStyle: List<String>.from(json['travel_style']),
        createdAt: DateTime.parse(json['created_at']),
        updatedAt: DateTime.parse(json['updated_at']),
      );

  Map<String, dynamic> toJson() => {
        'preferred_regions': preferredRegions,
        'travel_style': travelStyle,
        'created_at': createdAt.toIso8601String(),
        'updated_at': updatedAt.toIso8601String(),
      };
}

File: functions/entity/traveler.js
// travelers 컬렉션 (여행객 추가 정보)
// 문서 ID: userId
// {
//   preferred_regions: string[],
//   travel_style: string[],
//   created_at: timestamp,
//   updated_at: timestamp
// }

class Traveler {
  constructor(data) {
    this.preferred_regions = data.preferred_regions;
    this.travel_style = data.travel_style;
    this.created_at = data.created_at;
    this.updated_at = data.updated_at;
  }
}

module.exports = Traveler;
